src/Eshopworld.WorkerProcess/CosmosDistributedLock/DistributedLock.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace EShopworld.WorkerProcess.CosmosDistributedLock
+{
+    public class DistributedLock : IDistributedLock
+    {
+        private readonly ICosmosDistributedLockStore _cosmosDistributedLockStore;
+        private CosmosDistributedLockClaim _cosmosDistributedLockClaim;
+
+        public DistributedLock(ICosmosDistributedLockStore cosmosDistributedLockStore)
+        {
+            _cosmosDistributedLockStore = cosmosDistributedLockStore;
+        }
+
+        public async Task<IDisposable> Acquire(string lockName)
+        {
+            if (string.IsNullOrEmpty(lockName))
+                throw new ArgumentNullException(nameof(lockName));
+
+            try
+            {
+                await _cosmosDistributedLockStore.InitialiseAsync();
```

Comment:
Do we have to do it every time we acquire lock? I would prefer we do it once on the startup/initialisation - I am sure there must be cost associated to this execution that otherwise is unnecessary.

Reply:
I want to do it somewhere else but I don't know where else can I put it.

Reply:
You can do it as Lazy<> initialisation in the ctor for the lock store `_cosmosDistributedLockStore`.

Reply:
I instead changed it to lazy initialization of Document client in the `CosmosDistributedLockStore`. Since it is going to be a singleton.